src/instances/HeadcountInstance.ts

```diff
@@ -477,78 +482,90 @@ export class HeadcountInstance {
             this._guildDoc = res;
         }
 
-        this.stopAllIntervalsAndCollectors();
-        this.startControlPanelCollector();
+        this._logger.debug(`${this._instanceInfo} Stopping intervals and collectors`);   
 
         // End the collector since it's useless. We'll use it again though.
-        this.stopAllIntervalsAndCollectors("Headcount ended.");
+        await this.stopAllIntervalsAndCollectors("Headcount ended.");
 
         // Edit the control panel accordingly and re-react and start collector + intervals again.
+        this._logger.debug(`${this._instanceInfo} Replacing the headcount control panel`); 
         await this._controlPanelMsg.edit({
             embeds: [this.getControlPanelEmbed()!],
             components: HeadcountInstance.END_HEADCOUNT_BUTTONS
         }).catch();
+
+        this._logger.debug(`${this._instanceInfo} Restarting intervals and collectors`);  
         this.startControlPanelCollector();
         this.startIntervals();
 
         // Edit the headcount message
+        this._logger.debug(`${this._instanceInfo} Editing headcount embed`);  
         await this._headcountMsg.edit({
             embeds: [this.getHeadcountEmbed()!],
             content: "@here",
             components: AdvancedCollector.getActionRowsFromComponents(
                 this._afkCheckButtons.map(x => x.setDisabled(true))
             )
         }).catch();
+        this._logger.info(`${this._instanceInfo} Headcount ended`);
     }
 
 
     /**
      * Aborts a headcount
      */
     public async abortHeadcount(): Promise<void> {
-        console.info("Aborting headcount...");
+        this._logger.info(`${this._instanceInfo} Aborting headcount`);
         if (!this._headcountMsg || !this._controlPanelMsg)
             return;
 
         this._headcountStatus = HeadcountStatus.HEADCOUNT_ABORTED;
 
         // Stop 0: Stop all collectors
-        this.stopAllIntervalsAndCollectors("Headcount aborted.");
+        await this.stopAllIntervalsAndCollectors("Headcount aborted.");
         // Step 1: Remove from ActiveRaids collection
+        this._logger.debug(`${this._instanceInfo} Removing from active raids`);
         if (this._headcountMsg) {
             HeadcountInstance.ActiveHeadcounts.delete(this._headcountMsg.id);
         }
-
+        this._logger.debug(`${this._instanceInfo} Removing from database`);
         await Promise.all([
             // Step 2: Remove the raid object. We don't need it anymore.
             // Also stop all collectors.
             this.removeHeadcountFromDatabase(),
```

Comment:
If there's not two promises that needs to be resolved concurrently, just `await` this method.